server/src/navigation/messages.rs

```diff
@@ -39,14 +41,12 @@ pub mod navigation_query {
     impl Operation for Payload {
         type Output = Navigation;
 
-        async fn execute(&self, connection: Connection<'_, '_>) -> operation::Result<Self::Output> {
+        async fn execute<'e, A: sqlx::Acquire<'e, Database = sqlx::MySql> + std::marker::Send>(
+            &self,
+            acquire_from: A,
+        ) -> operation::Result<Self::Output> {
             let instance = self.instance.clone();
-            Ok(match connection {
-                Connection::Pool(pool) => Navigation::fetch(instance, pool).await?,
-                Connection::Transaction(transaction) => {
-                    Navigation::fetch_via_transaction(instance, transaction).await?
-                }
-            })
+            Ok(Navigation::fetch_via_transaction(instance, acquire_from).await?)
```

Comment:
Not so much left of those execute implementations. Could possibly be scratched all together at some point.